packages/plugin-avatar/test/unit/spec/avatar-url-store.js

```diff
@@ -0,0 +1,78 @@
+/**!
+ *
+ * Copyright (c) 2015-2016 Cisco Systems, Inc. See LICENSE file.
+ */
+import {assert} from '@ciscospark/test-helper-chai';
+import Avatar from '../../';
+import MockSpark from '@ciscospark/test-helper-mock-spark';
+
+describe(`Services`, () => {
+  describe(`Avatar`, () => {
+    describe(`AvatarUrlStore`, () => {
+      let store;
+      let spark;
+
+      beforeEach(() => {
+        spark = new MockSpark({
+          children: {
+            avatar: Avatar
+          }
+        });
+        store = spark.avatar.store;
+      });
+
+      it(`add item failures`, () => {
+        assert.isRejected(store.add(), `\`item\` is required`);
+        assert.isRejected(store.add({}), `\`uuid\` is required`);
+        assert.isRejected(store.add({uuid: `id1`}), `\`size\` is required`);
+        assert.isRejected(store.add({uuid: `id1`, size: 80}), `\`uuid\` does not appear to be a uuid`);
+        assert.isRejected(store.add({uuid: `id1@example.com`, size: 80}), `\`url\` is required`);
+        assert.isRejected(store.add({uuid: `id1@example.com`, size: 80, url: `http://www.example.com`}), `\`cacheControl\` is required`);
+      });
+      it(`get item failures`, () => {
+        assert.isRejected(store.get(), `\`item\` is required`);
+        assert.isRejected(store.get({}), `\`item.uuid\` is required`);
+        assert.isRejected(store.get({uuid: `id1`}), `\`item.size\` is required`);
+        assert.isRejected(store.get({uuid: `id1`, size: 80}), `\`item.uuid\` does not appear to be a uuid`);
+        assert.isRejected(store.get({uuid: `id1@example.com`, size: 80}), `No URL found by specified id`);
+      });
+      it(`set / get / remove`, () => {
+        const item1_40 = {uuid: `id1@example.com`, size: 40, url: `http://www-40.example.com`, cacheControl: 300};
+        const item1_50 = {uuid: `id1@example.com`, size: 50, url: `http://www-50.example.com`, cacheControl: 300};
+        const item1_80 = {uuid: `id1@example.com`, size: 80, url: `http://www-80.example.com`, cacheControl: 300};
+        const item1_110 = {uuid: `id1@example.com`, size: 110, url: `http://www-110.example.com`, cacheControl: 300};
+        const item1_135 = {uuid: `id1@example.com`, size: 135, url: `http://www-135.example.com`, cacheControl: 300};
+        const item1_192 = {uuid: `id1@example.com`, size: 192, url: `http://www-192.example.com`, cacheControl: 300};
+        const item1_640 = {uuid: `id1@example.com`, size: 640, url: `http://www-640.example.com`, cacheControl: 300};
+        const item1_1600 = {uuid: `id1@example.com`, size: 1600, url: `http://www-1600.example.com`, cacheControl: 300};
+        const item2_80 = {uuid: `id2@example.com`, size: 80, url: `http://www2.example.com`, cacheControl: 300};
+        const item3_80 = {uuid: `id3@example.com`, size: 80, url: `http://www3.example.com`, cacheControl: 300};
+
+        assert.isFulfilled(store.add(item1_80));
```

Comment:
`assert.isFulfilled`, `assert.isRejected`, `assert.eventually...`, are async, promise-returning functions, so they promises need to be returned at the end of the function.